test/compile_errors.zig

```diff
@@ -6979,6 +6979,17 @@ pub fn addCases(cases: *tests.CompileErrorContext) void {
         "tmp.zig:2:1: note: function declared here",
     });
 
+    cases.add("function pointer parameter is opaque",
+        \\const FooType = @OpaqueType();
+        \\export fn foo() void {
+        \\    const someFuncPtr: fn (FooType) void = undefined;
+        \\}
+    , &[_][]const u8{
+        "tmp.zig:3:28: error: parameter of opaque type 'FooType' not allowed",
+        "tmp.zig:3:24: note: referenced here",
+        "tmp.zig:3:44: note: referenced here",
+    });
+
```

Comment:
can you expand the compile-error test from 1-entry to 4-entries? something like this:
```zig
const FooType = @OpaqueType();
        
export fn entry1() void {
    const someFuncPtr: fn (FooType) void = undefined;
}               
                
export fn entry2() void {
    const someFuncPtr: fn (@TypeOf(null)) void = undefined;
}                       
                
export fn entry3() void {
    _ = foo1;
}       
        
export fn entry4() void {
    _ = foo2;
}           
            
fn foo1(foo: FooType) void {}
fn foo2(foo: @TypeOf(null)) void {}
```